pkg/utils/config/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	tomlv2 "github.com/pelletier/go-toml/v2"
+)
+
+const (
+	tomlFilePath = "/etc/containerd/config.toml"
+	nriPluginKey = "io.containerd.nri.v1.nri"
+	disableKey   = "disable"
+)
+
+func main() {
+	// Load TOML content
+	tomlData, err := os.ReadFile(tomlFilePath)
+	if err != nil {
+		log.Fatalf("Error reading TOML file: %v", err)
+	}
+
+	tomlMap, err := unmarshalTOML(tomlData)
+	if err != nil {
+		log.Fatalf("Error unmarshalling TOML: %v", err)
+	}
+
+	tomlMap = updateNRIplugin(tomlMap)
+
+	buf, err := addIndentation(tomlMap)
+	if err != nil {
+		log.Fatalf("Error adding indentation: %v", err)
+	}
+
+	if err := writeToFile(tomlFilePath, buf.String()); err != nil {
+		log.Fatalf("Error writing to file: %v", err)
+	}
+
+	if err := restartContainerd(); err != nil {
+		log.Fatalf("Failed to restard the containerd: %v", err)
+	}
+}
+
+// restartContainerd uses nsenter to access the host's mount and PID namespaces.
+// This enables us to restart the host's containerd from within the init container
+// after making necessary configuration modifications.
+func restartContainerd() error {
+	command := "nsenter"
```

Comment:
I still feel that we'd need a bigger/fatter comment/warning here/somewhere to clarify that this binary must be run in the host PID namespace (and consequently with `hostPID: true` when running as a Pod/daemonset) for any of this to work.

Reply:
we can integrate check that /proc/1/cmdline is present and have expected value (e.g. contains "systemd")

Reply:
I improved the comment here on the func() and also added a note in the documentation for the helm chart field.